app/services/get_pending_data_sharing_agreements_for_provider_user.rb

```diff
@@ -0,0 +1,9 @@
+class GetPendingDataSharingAgreementsForProviderUser
+  def self.call(provider_user:)
+    provider_user.providers.map { |provider|
+      if ProviderAgreement.data_sharing_agreements.for_provider(provider).empty?
+        ProviderAgreement.new(agreement_type: :data_sharing_agreement, provider: provider, provider_user: provider_user)
+      end
+    }.compact
```

Comment:
This executes a query per associated provider, can we avoid the N+1 somehow?

Reply:
yep, rewrote the query

Reply:
Got it down to one query, thanks!